lib/puppet/util/watchdog.rb

```diff
@@ -0,0 +1,35 @@
+module Puppet::Util::Watchdog
```

Comment:
I'm not the biggest fan of having this in another Util module. Could move it over to `Puppet::Util::Execution` or somewhere else that seems better for a "terminate child process after timeout" method.